### ISSUE TITLE
gffread crashes during tool load

### ISSUE THREAD

**Issue by bgruening** [galaxyproject/galaxy#980]
The tool `gffread` (Filters and/or converts GFF3/GTF2 records) from usegalaxy.org results in the following error.

```
Traceback (most recent call last):
  File "/usr/local/galaxy/galaxy-dist/lib/galaxy/web/framework/decorators.py", line 260, in decorator
    rval = func( self, trans, *args, **kwargs)
  File "/usr/local/galaxy/galaxy-dist/lib/galaxy/webapps/galaxy/api/tools.py", line 98, in build
    return tool.to_json(trans, kwd.get('inputs', kwd))
  File "/usr/local/galaxy/galaxy-dist/lib/galaxy/tools/__init__.py", line 2593, in to_json
    populate_state(trans, self.inputs, state_inputs, state_errors, params.__dict__)
  File "/usr/local/galaxy/galaxy-dist/lib/galaxy/tools/__init__.py", line 2462, in populate_state
    state[input.name] = input.get_initial_value(trans, context, history=history)
  File "/usr/local/galaxy/galaxy-dist/lib/galaxy/tools/parameters/grouping.py", line 619, in get_initial_value
    current_case = self.get_current_case( test_value, trans )
  File "/usr/local/galaxy/galaxy-dist/lib/galaxy/tools/parameters/grouping.py", line 569, in get_current_case
    raise ValueError( "No case matched value:", self.name, str_value )
ValueError: ('No case matched value:', 'gffs', None)

```

**Comment by martenson** [comment]
Still happens on Main. I assume @guerler would know the most about this API.

![screenshot 2016-01-18 11 04 42](https://cloud.githubusercontent.com/assets/1814954/12396051/5435a6d4-bdd3-11e5-8099-10c88dfe6615.png)

**Comment by guerler** [comment]
The test parameter of a conditional should not be optional. Removing 'optional="true"' fixes this issue.

`<param name="gff_fmt" type="select" optional="true" label="Feature File Output" help="(-o output.gff3|output.gtf)">`

**Comment by martenson** [comment]
@guerler so this fixes it then? https://github.com/galaxyproject/tools-devteam/pull/327

Planemo lints it fine, Galaxy installs it fine and then it throws an exception while loading because of an 'optional' flag? I think this should have failed earlier. 

ping @jmchilton

**Comment by jmchilton** [comment]
I'm not happy with the proposed solution - the new tool form breaks working tools. I can't get the optional flag to actually make conditionals optional on the old tool form - but the tool does work so it should work here. @guerler is it possible to just ignore this flag on the test parameters of conditionals using the new tool form?

**Comment by martenson** [comment]
This seems to be caused by change (fix even) in tool validation. However toolform should probably not be the first to implement such fix but rather the last. Planemo lint should be first, Tool Shed should be right after, Galaxy install framework should be the third and the toolform UI should follow.

This way the tool cascade works and following it will save us trouble.

**Comment by guerler** [comment]
Yeah, it looks like a validation thing and my preference would be to just fix the faulty xml. However, maybe another option is to force non-optional test parameters on parameter parsing, although it does not look great to me. I am saying this because in the future, we might actually allow optional conditional test parameters. https://github.com/galaxyproject/galaxy/compare/dev...guerler:fix_optional_000?expand=1.

**Comment by nsoranzo** [comment]
This was fixed in #1637.
